app/controllers/cards_controller.rb

```diff
@@ -50,14 +54,9 @@ def update
   end
 
   def move
-    List.move(
-      @card,
-      prev_item: @board.cards.find_by(id: params[:prev_id]),
-      next_item: @board.cards.find_by(id: params[:next_id])
-    )
-
-    if params[:new_list_id]
-      @card.list = @board.lists.find(params[:new_list_id])
+    List.move(@card, prev_item: @prev_item, next_item: @next_item)
+    if move_params[:new_list_id]
```

Comment:
This is a repeat of setting @parent below, we can DRY this up

Reply:
We cannot have both operations together in a single 'if' condition because we need to perform setting @parent before 'List.move' and update the new_list of a card after 'List.move' is performed. Performing setting @parent before 'List.move' is required because validate_move_params method is dependent on that. Updating the new_list of a card after 'List.move' is required to prevent breaking the linked list, which will happen when we move the head card of one list to another list and update the new_list is performed before 'List.move'. That is why we need to maintain this specific order and cannot DRY this up more. Please let me know if you have any suggestions.

Reply:
Your change of adding `new_list` looks good to me